src/impl/tcptransport.cpp

```diff
@@ -384,6 +384,7 @@ void TcpTransport::triggerBufferedAmount(size_t amount) {
 }
 
 void TcpTransport::process(PollService::Event event) {
+	auto self = weak_from_this().lock();
```

Comment:
You should return if `lock()` returns `nullptr`, just to be sure. 
```suggestion
	auto self = weak_from_this().lock();
	if (!self)
		return;
```